### PR TITLE
`[Move]` Part-7 Classes into Different Library - `Neo.Extensions`

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Description
Reorganized and move classes from `neo` library to `neo.extensions` library. 

### Fixed problems and warnings that occurred. Along with naming violations.

# Change Log
 - Moved methods from `Neo.VM.Helper` class
 - Added `ScriptBuilderExtensions` class
 - Added `using` statement
 - Fixed typo with `StringExtensions` class and filename
 - Fixed unit tests

## Type of change

- [x] Optimization (the change is only an optimization)
- [x] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- Unit Testing

image

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo.Extensions/Collections/HashSetExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            if (set.Count > other.Count)
+            {
+                set.ExceptWith(other);
+            }
+            else
+            {
+                set.RemoveWhere(u => other.Contains(u));
+            }
+        }
```
Comment (Hecate2) [inline src/Neo.Extensions/Collections/HashSetExtensions.cs:28]: ( Does `set.Remove(other)` have any difference from `set.ExceptWith(other)`, even when `set.Count <= other.Count`? )

Comment (cschuchardt88) [reply src/Neo.Extensions/Collections/HashSetExtensions.cs:28]: No clue. I'm just moving the method to appropriate namespaces or libraries.

**Comment by Jim8y** [comment]
@cschuchardt88 please fix conflics, is this one the current blocker?

**Comment by cschuchardt88** [comment]
Awaiting on `part-6` before I fix it

**Comment by Jim8y** [comment]
@cschuchardt88 please fix the conflicts.

**Comment by cschuchardt88** [comment]
@Jim8y done, fixed all merging errors

File: tests/Neo.Extensions.Tests/UT_BigIntegerExtensions.cs [hunk 2]
```diff
@@ -21,6 +21,7 @@ namespace Neo.Extensions.Tests
     [TestClass]
     public class UT_BigIntegerExtensions
     {
+
```
Comment (shargon) [inline tests/Neo.Extensions.Tests/UT_BigIntegerExtensions.cs:24]:
```suggestion
```

File: src/Neo/Extensions/VM/ScriptBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+                        builder.EmitPush((byte[])parameter.Value);
+                        break;
+                    case ContractParameterType.Boolean:
+                        builder.EmitPush((bool)parameter.Value);
+                        break;
+                    case ContractParameterType.Integer:
+                        if (parameter.Value is BigInteger bi)
+                            builder.EmitPush(bi);
+                        else
+                            builder.EmitPush((BigInteger)typeof(BigInteger).GetConstructor([parameter.Value.GetType()]).Invoke([parameter.Value]));
```
[the remark this answers is not in the document]
Comment (cschuchardt88) [reply src/Neo/Extensions/VM/ScriptBuilderExtensions.cs:190]: I rather leave it.

**Review by Jim8y — APPROVED** [review]

**Pull request merged**